src/components/_form-group.css

```diff
@@ -0,0 +1,122 @@
+/*
+
+  FORM GROUP
+
+  To combine inputs and buttons into a grouped utility.
+
+  Media Query Extensions:
+    -ns = not-small
+    -m  = medium
+    -l  = large
+
+*/
+
+.k--form-group {
+  display: inline-flex;
+}
+
+.k--form-group * {
+  position: relative;
+  z-index: 0;
```

Comment:
Do we really need a globally applied z-index?

z-index is usually pretty use-case-specific. Or is there some need directly in this component for layering?

Reply:
this is specifically for focused buttons in button groups. it ensures the focused button styles are applied on top of all the rest. Oh... but perhaps we don't need zero, and can just add `z-index` on the focused and active ones?

Reply:
ah, yeah it's needed because otherwise we'll get this

![image](https://user-images.githubusercontent.com/780079/29733777-bbb8e6de-89a3-11e7-9975-7d2a08c07dac.png)

Reply:
but we want this
![image](https://user-images.githubusercontent.com/780079/29733822-e826a274-89a3-11e7-9cf1-ff13aae9edc5.png)

Reply:
mmk